### PR TITLE
WIP: Add support for Decorator, which enables the operation/span name…

### PR DESCRIPTION
**Pull request by objectiser**
… to be specified

**What this PR does / why we need it**:
This PR enables a user to specify an operation name associated with the route, that can be used in the tracing operation/span name.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
This PR is currently using my fork of the istio/api repo, until https://github.com/istio/api/pull/186 is merged.

**Release note**:

```release-note
This enhancement will enable a specific tracing span/operation name to be associated with a route. This can be used to provide business relevant (meaningful) names for specific REST endpoints.
```

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -489,7 +489,7 @@ filegroup(
 )
     """,
     commit = ISTIO_API,
-    remote = "https://github.com/istio/api.git",
+    remote = "https://github.com/objectiser/istio-api.git",
```
Comment (linsun) [inline WORKSPACE]: are the 2 changes in this file intended?

File: model/conversion_test.go [hunk 2]
```diff
@@ -37,6 +37,9 @@ func TestProtoSchemaConversions(t *testing.T) {
 			{Destination: &proxyconfig.IstioService{Name: "bar"}, Weight: 75},
 			{Destination: &proxyconfig.IstioService{Name: "baz"}, Weight: 25},
 		},
+		Decorator: &proxyconfig.Decorator{
+			Operation: "foobar",
+		},
```
Comment (linsun) [inline model/conversion_test.go]: I assume decorator will change the microservice view in zipkin?  Does it impact Jaeger?  Would be nice to see some screen shot of it in zipkin or others.   Should we update our sample to use this in a follow-on PR?

File: WORKSPACE (hunk 1 above)
Comment (objectiser) [reply WORKSPACE]: This is temporary until the associated PR https://github.com/istio/api/pull/186 has been merged.

File: model/conversion_test.go (hunk 2 above)
Comment (objectiser) [reply model/conversion_test.go]: It changes the span name in the zipkin data - so will be displayed in any tracing system that is able to consume that format (e.g. jaeger). I'll see what I can do about a screenshot 😄 
Yes would be good to update the example - or atleast have some RouteRules that use it. But that can be a separate PR.

**Comment by objectiser** [comment]
istio-decorator

Looks like only the client span names are being set - so ok in jaeger, but inconsistent in zipkin:

istio-decorator-zipkin

So will take another look and see if I can get the server span names set.

**Comment by objectiser** [comment]
Closing this PR for now - requires further investigation so will reopen once have a more complete solution.

In the meantime I will create a separate PR for the zipkin test.

File: test/integration/zipkin.go [hunk 3]
```diff
@@ -85,8 +85,8 @@ func (t *zipkin) verifyTraces() error {
 		for _, id := range t.traces {
 			response := t.infra.clientRequest(
 				"t",
-				fmt.Sprintf("http://zipkin.%s:9411/api/v1/traces?annotationQuery=guid:x-request-id=/%s",
-					t.IstioNamespace, id),
+				fmt.Sprintf("http://zipkin.%s:9411/api/v1/traces",
```
[the remark this answers is not in the document]
Comment (objectiser) [reply test/integration/zipkin.go]: This fix has been submitted in a separate PR #1400 .
cc @rshriram @ijsnellf

**Pull request closed without merging**